boskos/janitor/janitor.py

```diff
@@ -27,36 +27,50 @@
-    Resource('compute', 'target-https-proxies', None, None, None, False, True),
-    Resource('compute', 'url-maps', None, None, None, False, True),
-    Resource('compute', 'backend-services', None, 'region', None, False, True),
-    Resource('compute', 'target-pools', None, 'region', None, False, True),
-    Resource('compute', 'health-checks', None, None, None, False, True),
-    Resource('compute', 'http-health-checks', None, None, None, False, True),
-    Resource('compute', 'instance-groups', None, 'zone', 'Yes', False, True),
-    Resource('compute', 'instance-groups', None, 'zone', 'No', False, True),
-    Resource('compute', 'instance-templates', None, None, None, False, True),
-    Resource('compute', 'networks', 'subnets', 'region', None, True, True),
-    Resource('compute', 'networks', None, '', None, False, True),
-    Resource('compute', 'routes', None, None, None, False, True),
+    Resource('', 'compute', 'instances', None, 'zone', None, False, True),
+    Resource('', 'compute', 'addresses', None, 'region', None, False, True),
+    Resource('', 'compute', 'disks', None, 'zone', None, False, True),
+    Resource('', 'compute', 'firewall-rules', None, None, None, False, True),
+    Resource('', 'compute', 'routes', None, None, None, False, True),
+    Resource('', 'compute', 'forwarding-rules', None, 'region', None, False, True),
+    Resource('', 'compute', 'target-http-proxies', None, None, None, False, True),
+    Resource('', 'compute', 'target-https-proxies', None, None, None, False, True),
+    Resource('', 'compute', 'url-maps', None, None, None, False, True),
+    Resource('', 'compute', 'backend-services', None, 'region', None, False, True),
+    Resource('', 'compute', 'target-pools', None, 'region', None, False, True),
+    Resource('', 'compute', 'health-checks', None, None, None, False, True),
+    Resource('', 'compute', 'http-health-checks', None, None, None, False, True),
+    Resource('', 'compute', 'instance-groups', None, 'zone', 'Yes', False, True),
+    Resource('', 'compute', 'instance-groups', None, 'zone', 'No', False, True),
+    Resource('', 'compute', 'instance-templates', None, None, None, False, True),
+    Resource('alpha', 'compute', 'network-endpoint-groups', None, None, None, False, True),
+    Resource('', 'compute', 'networks', 'subnets', 'region', None, True, True),
+    Resource('', 'compute', 'networks', None, '', None, False, True),
+    Resource('', 'compute', 'routes', None, None, None, False, True),
 
     # logging resources
-    Resource('logging', 'sinks', None, None, None, False, False),
+    Resource('', 'logging', 'sinks', None, None, None, False, False),
 ]
 
+def base_command(api_version):
+    """ Return the base gcloud command with api_version if exists.
+
+    Args:
+        api_version: the api version of the gcloud.
+    Returns:
+        list of base commands of gcloud .
+    """
+
+    base = ['gcloud']
+    if api_version:
+        cmd += [api_version]
```

Comment:
base

Reply:
probably worth take in resource, so we can have group, name and subgroup all handled here